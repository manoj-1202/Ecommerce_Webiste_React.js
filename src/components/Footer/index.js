import { Grid, Typography } from "@mui/material";
import { Link } from "react-router-dom";
import { FooterContainer, FooterTitle, FooterList, FooterListItemText, FooterSocialIcons, FooterEmailContainer, FooterListItemButton } from "../../styles/FooterStyles";
import FacebookIcon from "@mui/icons-material/Facebook";
import TwitterIcon from "@mui/icons-material/Twitter";
import InstagramIcon from "@mui/icons-material/Instagram";
import { Email } from "@mui/icons-material";

export default function Footer() {
  return (
    <FooterContainer>
      <Grid container spacing={1} justifyContent="space-around">
        <Grid item md={3} lg={4}>
          <FooterTitle variant="body1">About us</FooterTitle>
          <Typography variant="caption2">
            We blend style and functionality with our premium bag collection. From luxurious leather to versatile travel options, each piece is crafted with meticulous attention to detail.
          </Typography>
        </Grid>
        <Grid item md={3} lg={2}>
          <FooterTitle variant="body1">Information</FooterTitle>
          <FooterList>
            <FooterListItemText>
              <Typography variant="caption2">About Us</Typography>
            </FooterListItemText>
            <FooterListItemText>
              <Typography variant="caption2">Order Tracking</Typography>
            </FooterListItemText>
            <FooterListItemText>
              <Typography variant="caption2">Privacy &amp; Policy</Typography>
            </FooterListItemText>
            <FooterListItemText>
              <Typography variant="caption2">Terms &amp; Conditions</Typography>
            </FooterListItemText>
          </FooterList>
        </Grid>
        <Grid item md={3} lg={2}>
          <FooterTitle variant="body1">My Account</FooterTitle>
          <FooterList>
            <FooterListItemButton component={Link} to="/login">
              <FooterListItemText primary="LogIn" />
            </FooterListItemButton>
            <FooterListItemButton component={Link} to="/signup">
              <FooterListItemText primary="SignUp" />
            </FooterListItemButton>
            <FooterListItemButton component={Link} to="/cart">
              <FooterListItemText primary="Cart" />
            </FooterListItemButton>
            <FooterListItemButton component={Link} to="/favorites">
              <FooterListItemText primary="Favorites" />
            </FooterListItemButton>
          </FooterList>
        </Grid>

        <Grid item md={3} lg={4}>
          <FooterTitle variant="body1">Mail Us</FooterTitle>
          <FooterEmailContainer>
            <Email sx={{ marginRight: 1 }} />
            <Typography>mybags@gmail.com</Typography>
          </FooterEmailContainer>
          <FooterSocialIcons>
            <FacebookIcon sx={{ mr: 3 }} />
            <TwitterIcon sx={{ mr: 3 }} />
            <InstagramIcon />
          </FooterSocialIcons>
        </Grid>
      </Grid>
    </FooterContainer>
  );
}
