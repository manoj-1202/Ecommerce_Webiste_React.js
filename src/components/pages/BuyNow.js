import React from 'react';
import { Container, Typography, Card, CardContent, Box, Grid } from '@mui/material';
import { useCart } from '../products/CartContext';
import { useNavigate } from 'react-router-dom';
import { Euro, ShoppingCart, LocalOffer } from '@mui/icons-material';
import { ProductRow, StyledBox, StyledProductBox, StyledImage, StyledButton, StyledTypography } from '../../styles/productsStyles/buyNowStyles';

const BuyNow = () => {
  const { cart, getTotalAmount } = useCart();
  const navigate = useNavigate();

  const handlePlaceOrder = () => {
    const totalAmount = getTotalAmount();
    const orderDetails = {
      date: new Date().toLocaleDateString(),
      totalAmount,
      products: cart,
    };
    const userId = localStorage.getItem('currentUser');
    localStorage.setItem(`orderDetails-${userId}`, JSON.stringify(orderDetails));
    navigate('/payment', { state: { totalAmount } });
  };

  return (
    <Container sx={{ marginBottom: 4 }}>
      <StyledTypography variant="h4" gutterBottom align="center">
        Place Your Order
      </StyledTypography>
      {cart.length > 0 ? (
        <Card variant="outlined" sx={{ margin: 'auto', padding: '10px', maxWidth: 800, boxShadow: 3 }}>
          <CardContent>
            <ProductRow>
              {cart.map((product) => (
                <StyledProductBox key={product.id}>
                  <Grid container spacing={2}>
                    <Grid item xs={12} sm={6}>
                      <StyledImage src={product.image} alt={product.name} />
                    </Grid>
                    <Grid item xs={12} sm={6}>
                      <Typography variant="h6" gutterBottom>{product.name}</Typography>
                      <Typography variant="body2" color="text.secondary">{product.description}</Typography>
                      <Box display="flex" alignItems="center" mb={1}>
                        <Euro sx={{ mr: 1 }} />
                        <Typography variant="body2" color="text.primary">Price: ₹{product.price}</Typography>
                      </Box>
                      <Box display="flex" alignItems="center" mb={1}>
                        <ShoppingCart sx={{ mr: 1 }} />
                        <Typography variant="body2" color="text.primary">Quantity: {product.quantity}</Typography>
                      </Box>
                      <Box display="flex" alignItems="center">
                        <LocalOffer sx={{ mr: 1 }} />
                        <Typography variant="body2" color="text.primary">Total: ₹{(product.price * product.quantity).toFixed(2)}</Typography>
                      </Box>
                    </Grid>
                  </Grid>
                </StyledProductBox>
              ))}
            </ProductRow>
            <StyledBox>
              <Typography variant="body1" display="flex" alignItems="center" fontWeight={"bold"}>
                Total Amount: ₹ {getTotalAmount()}
              </Typography>
              <StyledButton onClick={handlePlaceOrder}>
                Buy Now
              </StyledButton>
            </StyledBox>
          </CardContent>
        </Card>
      ) : (
        <Typography variant="h6" align="center">
          No items in the cart.
        </Typography>
      )}
    </Container>
  );
};

export default BuyNow;
