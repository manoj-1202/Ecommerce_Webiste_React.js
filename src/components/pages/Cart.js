import React from 'react';
import { useCart } from '../products/CartContext';
import { Container, Typography, Grid } from '@mui/material';
import { CartProduct, ProductImage, CartQuantityBox, CartButton, RemoveButton, TotalAmountContainer, TotalAmountText, BuyNow } from '../../styles/productsStyles/cartStyles';
import ProductMeta from '../products/ProductMeta';
import { useNavigate } from 'react-router-dom';

const formatPrice = (price) => `â‚¹${price}`;

const Cart = () => {
  const { cart, increaseQuantity, decreaseQuantity, removeFromCart, getTotalAmount } = useCart();
  const navigate = useNavigate();

  const handleIncreaseQuantity = (productId) => {
    increaseQuantity(productId);
  };

  const handleDecreaseQuantity = (productId) => {
    decreaseQuantity(productId);
  };

  const handleRemoveFromCart = (productId) => {
    removeFromCart(productId);
  };

  const handleBuyNow = () => {
    navigate('/buynow');
  };

  return (
    <Container sx={{ marginBottom: 1 }}>
      <Typography variant="h4" gutterBottom align="center">
        Your Cart
      </Typography>
      {cart.length > 0 ? (
        <>
          <Grid container spacing={1}>
            {cart.map((product) => (
              <Grid item xs={12} sm={6} md={3} key={product.id}>
                <CartProduct>
                  <ProductImage src={product.image} alt={product.name} />
                  <ProductMeta product={{ ...product, price: formatPrice(product.price) }} />
                  <CartQuantityBox>
                    <CartButton
                      variant="contained"
                      color="success"
                      onClick={() => handleDecreaseQuantity(product.id)}
                    >
                      Less
                    </CartButton>
                    <Typography variant="body2" sx={{ marginX: 1 }}>
                      Quantity: {product.quantity}
                    </Typography>
                    <CartButton
                      variant="contained"
                      color="success"
                      onClick={() => handleIncreaseQuantity(product.id)}
                    >
                      Add
                    </CartButton>
                  </CartQuantityBox>
                  <RemoveButton
                    variant="contained"
                    onClick={() => handleRemoveFromCart(product.id)}
                  >
                    Remove
                  </RemoveButton>
                </CartProduct>
              </Grid>
            ))}
          </Grid>
          <TotalAmountContainer>
            <TotalAmountText variant="h5" align="center">
              Total Amount: {formatPrice(getTotalAmount())}
            </TotalAmountText>
            <BuyNow onClick={handleBuyNow}>
              Place Order
            </BuyNow>
          </TotalAmountContainer>
        </>
      ) : (
        <Typography variant="h6" align="center">
          No items in the cart.
        </Typography>
      )}
    </Container>
  );
};

export default Cart;
