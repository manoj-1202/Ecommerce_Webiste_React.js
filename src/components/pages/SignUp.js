import React, { useState } from 'react';
import { Avatar, Button, CssBaseline, TextField, Link, Grid, Box, Typography, Container, Snackbar, Alert } from '@mui/material';
import LockOutlinedIcon from '@mui/icons-material/LockOutlined';
import { createTheme, ThemeProvider } from '@mui/material/styles';
import { useNavigate } from 'react-router-dom';

const defaultTheme = createTheme();

export default function SignUp() {
  const navigate = useNavigate();
  const [open, setOpen] = useState(false);
  const [message, setMessage] = useState('');
  const [severity, setSeverity] = useState('success');

  const handleSubmit = (event) => {
    event.preventDefault();
    const data = new FormData(event.currentTarget);

    const user = {
      firstName: data.get('firstName'),
      lastName: data.get('lastName'),
      email: data.get('email'),
      password: data.get('password'),
    };

    const users = JSON.parse(localStorage.getItem('users')) || [];
    if (users.find(u => u.email === user.email)) {
      setMessage('User with this email already exists. Please log in.');
      setSeverity('error');
      setOpen(true);
      return;
    }

    users.push(user);
    localStorage.setItem('users', JSON.stringify(users));

    setMessage('Sign up successful! You can now log in.');
    setSeverity('success');
    setOpen(true);

    setTimeout(() => {
      navigate('/login');
    }, 1000);
  };

  const handleClose = (event, reason) => {
    if (reason === 'clickaway') {
      return;
    }
    setOpen(false);
  };

  return (
    <ThemeProvider theme={defaultTheme}>
      <Container component="main" maxWidth="xs">
        <CssBaseline />
        <Box sx={{ display: 'flex', flexDirection: 'column', alignItems: 'center' }}>
          <Avatar sx={{ m: 1, bgcolor: 'secondary.main' }}>
            <LockOutlinedIcon />
          </Avatar>
          <Typography component="h1" variant="h5">
            Sign up
          </Typography>
          <Box component="form" noValidate onSubmit={handleSubmit} sx={{ mt: 3 }}>
            <Grid container spacing={2}>
              <Grid item xs={12} sm={6}>
                <TextField
                  autoComplete="given-name"
                  name="firstName"
                  required
                  fullWidth
                  id="firstName"
                  label="First Name"
                  autoFocus
                />
              </Grid>
              <Grid item xs={12} sm={6}>
                <TextField
                  required
                  fullWidth
                  id="lastName"
                  label="Last Name"
                  name="lastName"
                  autoComplete="family-name"
                />
              </Grid>
              <Grid item xs={12}>
                <TextField
                  required
                  fullWidth
                  id="email"
                  label="Email Address"
                  name="email"
                  autoComplete="email"
                />
              </Grid>
              <Grid item xs={12}>
                <TextField
                  required
                  fullWidth
                  name="password"
                  label="Password"
                  type="password"
                  id="password"
                  autoComplete="new-password"
                />
              </Grid>
            </Grid>
            <Button
              type="submit"
              fullWidth
              variant="contained"
              sx={{ mt: 3, mb: 2 }}
            >
              Sign Up
            </Button>
            <Grid container justifyContent="flex-end">
              <Grid item>
                <Link href="/login" variant="body2">
                  Already have an account? Log In
                </Link>
              </Grid>
            </Grid>
          </Box>
        </Box>
        <Snackbar open={open} autoHideDuration={5000} onClose={handleClose}
          anchorOrigin={{ vertical: 'top', horizontal: 'center' }}>
          <Alert onClose={handleClose} severity={severity} sx={{ width: '100%' }}>
            {message}
          </Alert>
        </Snackbar>
      </Container>
    </ThemeProvider>
  );
}
