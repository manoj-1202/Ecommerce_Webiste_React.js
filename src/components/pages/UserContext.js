import React, { createContext, useContext, useState, useEffect } from 'react';
import { Alert, Snackbar } from '@mui/material';

const UserContext = createContext();

export const UserProvider = ({ children }) => {
  const [user, setUser] = useState(null);
  const [snackbar, setSnackbar] = useState({ open: false, message: '', severity: 'info' });

  useEffect(() => {
    const currentUserEmail = sessionStorage.getItem('currentUser') || localStorage.getItem('currentUser');
    if (currentUserEmail) {
      const userData = JSON.parse(sessionStorage.getItem(currentUserEmail) || localStorage.getItem(currentUserEmail));
      if (userData) {
        setUser(userData);
      }
    }
  }, []);

  const login = (userData) => {
    setUser(userData);
    sessionStorage.setItem('currentUser', userData.email);
    sessionStorage.setItem(userData.email, JSON.stringify(userData));
    localStorage.setItem('currentUser', userData.email);
    localStorage.setItem(userData.email, JSON.stringify(userData));
    showAlert('Login successful!', 'success');
  };

  const logout = () => {
    const currentUserEmail = sessionStorage.getItem('currentUser') || localStorage.getItem('currentUser');
    if (currentUserEmail) {
      sessionStorage.removeItem('currentUser');
      sessionStorage.removeItem(currentUserEmail);
      localStorage.removeItem('currentUser');
      localStorage.removeItem(currentUserEmail);
    }
    setUser(null);
    showAlert('Logout successful!', 'info');
  };

  const showAlert = (message, severity = 'info') => {
    setSnackbar({ open: true, message, severity });
  };

  const handleCloseSnackbar = () => {
    setSnackbar({ ...snackbar, open: false });
  };

  return (
    <UserContext.Provider value={{ user, login, logout, showAlert }}>
      {children}
      <Snackbar
        open={snackbar.open}
        autoHideDuration={6000}
        onClose={handleCloseSnackbar}
        anchorOrigin={{ vertical: 'center', horizontal: 'center' }}
      >
        <Alert onClose={handleCloseSnackbar} severity={snackbar.severity}>
          {snackbar.message}
        </Alert>
      </Snackbar>
    </UserContext.Provider>
  );
};

export const useUser = () => useContext(UserContext);
