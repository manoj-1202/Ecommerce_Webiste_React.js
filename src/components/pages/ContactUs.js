import React, { useState } from 'react';
import { Grid, Box, Typography, TextField, Button, Container, Snackbar, Alert } from '@mui/material';
import { LocationOn, Phone, Email } from '@mui/icons-material';
import emailjs from 'emailjs-com';
import { useUser } from './UserContext'; 

const ContactUs = () => {
  const { user } = useUser(); 
  const [open, setOpen] = useState(false);
  const [formData, setFormData] = useState({ name: '', email: '', message: '' });
  const [errorOpen, setErrorOpen] = useState(false);
  const [notLoggedInOpen, setNotLoggedInOpen] = useState(false); 

  const handleChange = (event) => {
    const { name, value } = event.target;
    setFormData({ ...formData, [name]: value });
  };

  const handleSubmit = (event) => {
    event.preventDefault();

    if (!user) {
      setNotLoggedInOpen(true);
      return;
    }

    console.log("Sending the following data:", formData); 

    emailjs.send(
      'service_hq3cbqm',  //service id
      'template_3cb77ma', // template id
      formData,
      'asDhf4VWP04Z1Qc7J' //api id
    ).then((result) => {
      console.log('Email successfully sent!', result);
      setOpen(true);
      setFormData({ name: '', email: '', message: '' }); 
    }, (error) => {
      console.error('Failed to send email:', error);
      setErrorOpen(true);
    });
  };

  const handleClose = (reason) => {
    if (reason === 'clickaway') {
      return;
    }
    setOpen(false);
    setErrorOpen(false);
    setNotLoggedInOpen(false); 
  };

  return (
    <Container maxWidth="lg" sx={{ marginBottom: 4 }}>
      <Grid container spacing={4} sx={{ marginTop: 4 }}>
        <Grid item xs={12} md={6}>
          <Box sx={{ backgroundColor: '#f5f5f5', padding: 4 }}>
            <Typography variant="h4" gutterBottom>
              Contact Us
            </Typography>
            <Typography variant="body1" paragraph>
              We're here to answer your questions and discuss the decentralized future of the internet.
            </Typography>
            <Box sx={{ display: 'flex', alignItems: 'center', marginBottom: 2 }}>
              <LocationOn sx={{ marginRight: 1 }} />
              <Typography variant="body2">My Bags , Coimbatore , 641002 , Tamil Nadu</Typography>
            </Box>
            <Box sx={{ display: 'flex', alignItems: 'center', marginBottom: 2 }}>
              <Phone sx={{ marginRight: 1 }} />
              <Typography variant="body2">+91 555-333-555</Typography>
            </Box>
            <Box sx={{ display: 'flex', alignItems: 'center' }}>
              <Email sx={{ marginRight: 1 }} />
              <Typography variant="body2">mybags@gmail.com</Typography>
            </Box>
          </Box>
        </Grid>
        
        <Grid item xs={12} md={6}>
        
          <Box
            component="form"
            sx={{ display: 'flex', flexDirection: 'column', gap: 1 }}
            onSubmit={handleSubmit}
          >
            <Typography variant='h5'>Feedback :</Typography>
            <TextField
              required
              id="name"
              name="name"
              label="Your Name"
              variant="outlined"
              fullWidth
              value={formData.name}
              onChange={handleChange}
            />
            <TextField
              required
              id="email"
              name="email"
              label="Email Address"
              variant="outlined"
              fullWidth
              value={formData.email}
              onChange={handleChange}
            />
            <TextField
              required
              id="message"
              name="message"
              label="Feedback"
              variant="outlined"
              multiline
              rows={4}
              fullWidth
              value={formData.message}
              onChange={handleChange}
            />
            <Button type="submit" variant="contained" sx={{ backgroundColor: 'blue' }}>
              Submit
            </Button>
          </Box>
        </Grid>
      </Grid>

      <Snackbar open={open} autoHideDuration={5000} onClose={handleClose}>
        <Alert onClose={handleClose} severity="success" sx={{ width: '100%' }}>
          Your message has been sent!
        </Alert>
      </Snackbar>

      <Snackbar open={errorOpen} autoHideDuration={5000} onClose={handleClose}>
        <Alert onClose={handleClose} severity="error" sx={{ width: '100%' }}>
          Failed to send your message. Please try again later.
        </Alert>
      </Snackbar>

      <Snackbar open={notLoggedInOpen} autoHideDuration={5000} onClose={handleClose}>
        <Alert onClose={handleClose} severity="warning" sx={{ width: '100%' }}>
          You need to log in to send a Feedback.
        </Alert>
      </Snackbar>
    </Container>
  );
};

export default ContactUs;
