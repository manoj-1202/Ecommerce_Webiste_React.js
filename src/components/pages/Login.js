import React, { useState } from 'react';
import { Avatar, Button, CssBaseline, TextField, Link, Grid, Box, Typography, Container, Snackbar, Alert } from '@mui/material';
import LockOutlinedIcon from '@mui/icons-material/LockOutlined';
import { createTheme, ThemeProvider } from '@mui/material/styles';
import { useUser } from './UserContext';
import { useNavigate } from 'react-router-dom';

const defaultTheme = createTheme();

export default function Login() {
  const [error, setError] = useState(null);
  const [openSnackbar, setOpenSnackbar] = useState(false);
  const [snackbarMessage, setSnackbarMessage] = useState('');
  const { login } = useUser();
  const navigate = useNavigate();

  const handleSubmit = (event) => {
    event.preventDefault();
    const data = new FormData(event.currentTarget);
    const email = data.get('email');
    const password = data.get('password');


    const users = JSON.parse(localStorage.getItem('users')) || [];
    const user = users.find(u => u.email === email && u.password === password);

    if (user) {
      const token = btoa(`${email}:${password}`);
      const userData = { ...user, token };

      localStorage.setItem('currentUser', user.id);
      sessionStorage.setItem('currentUser', user.id);
      sessionStorage.setItem(user.id, JSON.stringify(userData));

      login(userData);
      setSnackbarMessage('Login successful!');
      setOpenSnackbar(true);

      // Use navigate function properly
      setTimeout(() => {
        console.log('Navigating to /login');
        navigate('/');
      }, 1000);
    } else {
      setError('Invalid email or password');
    }
  };

  const handleCloseSnackbar = () => {
    setOpenSnackbar(false);
  };


  return (
    <ThemeProvider theme={defaultTheme}>
      <Container component="main" maxWidth="xs">
        <CssBaseline />
        <Box sx={{ display: 'flex', flexDirection: 'column', alignItems: 'center' }}>
          <Avatar sx={{ m: 1, bgcolor: 'secondary.main' }}>
            <LockOutlinedIcon />
          </Avatar>
          <Typography component="h1" variant="h5">
            Log In
          </Typography>
          <Box component="form" onSubmit={handleSubmit} noValidate sx={{ mt: 1 }}>
            <TextField
              margin="normal"
              required
              fullWidth
              id="email"
              label="Email Address"
              name="email"
              autoComplete="email"
              autoFocus
            />
            <TextField
              margin="normal"
              required
              fullWidth
              name="password"
              label="Password"
              type="password"
              id="password"
              autoComplete="current-password"
            />
            {error && (
              <Typography color="error" variant="body2">
                {error}
              </Typography>
            )}
            <Button type="submit" fullWidth variant="contained" sx={{ mt: 3, mb: 2 }}>
              Sign In
            </Button>
            <Grid container>
              <Grid item xs>
              </Grid>
              <Grid item>
                <Link href="/signup" variant="body2">
                  {"Don't have an account? Sign Up"}
                </Link>
              </Grid>
            </Grid>
          </Box>
        </Box>

        {/* Snackbar for login success */}
        <Snackbar
          open={openSnackbar}
          autoHideDuration={6000}
          onClose={handleCloseSnackbar}
          anchorOrigin={{ vertical: 'top', horizontal: 'center' }}
        >
          <Alert onClose={handleCloseSnackbar} severity="success" sx={{ width: '100%' }}>
            {snackbarMessage}
          </Alert>
        </Snackbar>
      </Container>
    </ThemeProvider>
  );
}
