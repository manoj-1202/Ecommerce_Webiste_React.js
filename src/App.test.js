import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import '@testing-library/jest-dom';
import { create } from 'react-test-renderer';
import { ThemeProvider } from '@mui/material';
import theme from './styles/theme';
import Banner from './components/banner';
import BuyNow from "./components/pages/BuyNow";
import Cart from './components/pages/Cart';
import ContactUs from './components/pages/ContactUs';
import Footer from "./components/Footer";
import Login from './components/pages/Login';
import PaymentOptions from './components/pages/Payment';
import SignUp from './components/pages/SignUp';
import OrderCard from "./components/pages/OrderDetails";
import { useNavigate } from 'react-router-dom';
import { CartProvider } from './components/products/CartContext';
import { useUser } from './components/pages/UserContext';

// Mock useNavigate
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: jest.fn(),
}));

// Mock the useCart hook and CartProvider
jest.mock('./components/products/CartContext', () => ({
    CartProvider: ({ children }) => <div>{children}</div>,
    useCart: () => ({
        cart: [{ id: '1', name: 'Test Product', price: 100, quantity: 1, image: 'test.jpg', description: 'Test Description' }],
        getTotalAmount: () => 100,
    })
}));

// Mock the useUser hook
jest.mock('./components/pages/UserContext', () => ({
    useUser: jest.fn(),
}));

// Mock useNavigate
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: jest.fn(),
}));

describe('App Component Tests', () => {

    // Test for Banner component
    test('renders Banner component without crashing', () => {
        render(
            <Router>
                <Banner />
            </Router>
        );
    });

    test('matches snapshot for Banner component', () => {
        const component = create(
            <Router>
                <Banner />
            </Router>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('navigates to products page on button click in Banner', () => {
        const navigate = jest.fn(); 
        useNavigate.mockReturnValue(navigate); 

        render(
            <Router>
                <Banner />
            </Router>
        );

        const button = screen.getByText(/Shop Now/i);
        fireEvent.click(button);
        expect(navigate).toHaveBeenCalledWith('/products');
    });

    // Test for BuyNow component
    test('navigates to payment page on button click in BuyNow', () => {
        const navigate = jest.fn(); 
        useNavigate.mockReturnValue(navigate); 

        render(
            <Router>
                <ThemeProvider theme={theme}>
                    <CartProvider>
                        <BuyNow />
                    </CartProvider>
                </ThemeProvider>
            </Router>
        );

        const button = screen.getByText(/Buy Now/i);
        fireEvent.click(button);
        expect(navigate).toHaveBeenCalledWith('/payment', expect.any(Object)); 
    });

    jest.mock('@mui/system', () => ({
        ...jest.requireActual('@mui/system'),
        useTheme: () => ({
          spacing: jest.fn().mockImplementation((factor) => factor * 8),
        }),
      }));
      

    test('renders BuyNow component without crashing', () => {
        render(
            <Router>
                <ThemeProvider theme={theme}>
                    <CartProvider>
                        <BuyNow />
                    </CartProvider>
                </ThemeProvider>
            </Router>
        );

        const button = screen.getByText(/Buy Now/i);
        expect(button).toBeInTheDocument();
    });

    // Test for Cart component
    jest.mock('./components/pages/UserContext', () => ({
        useUser: jest.fn(),
      }));
      
      const MockCartProvider = ({ children }) => {
        const mockCartValue = {
          cartItems: [
            { id: '1', name: 'Sample Product', quantity: 1, totalAmount: 100 }
          ]
        };
      
        return (
          <CartProvider value={mockCartValue}>
            {children}
          </CartProvider>
        );
      };
      
      describe('Cart Component', () => {
        test('should display product added to the cart', async () => {
          const mockUser = { email: 'test@example.com' };
          useUser.mockReturnValue({ user: mockUser });
      
          render(
            <Router>
                <ThemeProvider theme={theme}>
              <MockCartProvider>
                <Cart />
                
              </MockCartProvider>
              </ThemeProvider>
            </Router>
          );
      
          // Debug output to verify rendering
          screen.debug();
      
         
        });
      });
      
      
                       // Test for ContactUs component
    test('renders ContactUs component without crashing', () => {
        useUser.mockReturnValue({
            user: { id: 'testUser', name: 'Test User', email: 'test@example.com' }
        });

        render(
            <Router>
                <ContactUs />
            </Router>
        );
        
 fireEvent.change(screen.getByLabelText(/Your Name/i), { target: { value: 'John Doe' } });
 fireEvent.change(screen.getByLabelText(/Email Address/i), { target: { value: 'john@example.com' } });
 fireEvent.change(screen.getByLabelText(/Feedback/i), { target: { value: 'Great service!' } });

 expect(screen.getByRole('button', { name: /submit/i })).toBeInTheDocument();
 
    });

                          // Test for Footer component
    test('renders Footer component without crashing', () => {
        render(
            <Router>
                <Footer />
            </Router>
        );
    });



                       // Test for Login component

    jest.mock('./components/pages/UserContext', () => ({
      useUser: jest.fn(),
    }));
    
    const mockLogin = jest.fn();
    
    describe('Login Component', () => {
      beforeEach(() => {
        useUser.mockReturnValue({ login: mockLogin });
        // Setup initial localStorage state
        localStorage.setItem('users', JSON.stringify([{ id: '1', email: 'test@example.com', password: 'password' }]));
        localStorage.setItem('currentUser', null);
        sessionStorage.clear();
      });
    
      afterEach(() => {
        jest.clearAllMocks();
        localStorage.clear();
      });
    
      test('displays an error message with invalid credentials', async () => {
        render(
          <Router>
            <Login />
          </Router>
        );
    
        fireEvent.change(screen.getByLabelText(/email/i), { target: { value: 'wrong@example.com' } });
        fireEvent.change(screen.getByLabelText(/password/i), { target: { value: 'wrongpassword' } });
        fireEvent.click(screen.getByRole('button', { name: /sign in/i }));
    
        await waitFor(() => {
          expect(screen.getByText(/invalid email or password/i)).toBeInTheDocument();
        });
      });
    
      test('logs in successfully with correct credentials', async () => {
        render(
          <Router>
            <Login />
          </Router>
        );
    
        fireEvent.change(screen.getByLabelText(/email/i), { target: { value: 'test@example.com' } });
        fireEvent.change(screen.getByLabelText(/password/i), { target: { value: 'password' } });
        fireEvent.click(screen.getByRole('button', { name: /sign in/i }));
    
        // Check if the login function was called with the correct user data
        await waitFor(() => {
          expect(mockLogin).toHaveBeenCalledWith({
            id: '1',
            email: 'test@example.com',
            password: 'password',
            token: 'dGVzdEBleGFtcGxlLmNvbTpwYXNzd29yZA==', // Base64 encoded email:password
          });
        });
    
        // Check if the snackbar message appears
        expect(screen.getByText(/login successful/i)).toBeInTheDocument();
      });
    });
    
      

  
                          // Test for SignUp component

    
   describe('SignUp Component', () => {
  beforeEach(() => {
    // Clear localStorage before each test
    localStorage.clear();
  });

  test('renders SignUp form', () => {
    render(
      <Router>
        <SignUp />
      </Router>
    );

    expect(screen.getByLabelText(/First Name/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/Last Name/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/Email Address/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/Password/i)).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /Sign Up/i })).toBeInTheDocument();
  });

  test('shows error when user already exists', async () => {
    // Set up an existing user in localStorage
    const users = [{ email: 'existing@example.com', password: 'password123' }];
    localStorage.setItem('users', JSON.stringify(users));

    render(
      <Router>
        <SignUp />
      </Router>
    );

    fireEvent.change(screen.getByLabelText(/Email Address/i), { target: { value: 'existing@example.com' } });
    fireEvent.change(screen.getByLabelText(/Password/i), { target: { value: 'password123' } });
    fireEvent.change(screen.getByLabelText(/First Name/i), { target: { value: 'John' } });
    fireEvent.change(screen.getByLabelText(/Last Name/i), { target: { value: 'Doe' } });
    fireEvent.click(screen.getByRole('button', { name: /Sign Up/i }));

    await waitFor(() => {
      expect(screen.getByText(/User with this email already exists/i)).toBeInTheDocument();
    });
  });

  test('signs up new user successfully', async () => {
    render(
      <Router>
        <SignUp />
      </Router>
    );

    fireEvent.change(screen.getByLabelText(/First Name/i), { target: { value: 'John' } });
    fireEvent.change(screen.getByLabelText(/Last Name/i), { target: { value: 'Doe' } });
    fireEvent.change(screen.getByLabelText(/Email Address/i), { target: { value: 'john.doe@example.com' } });
    fireEvent.change(screen.getByLabelText(/Password/i), { target: { value: 'password123' } });
    fireEvent.click(screen.getByRole('button', { name: /Sign Up/i }));

    await waitFor(() => {
      expect(screen.getByText(/Sign up successful/i)).toBeInTheDocument();
    });

    // Check that the user is added to localStorage
    const users = JSON.parse(localStorage.getItem('users'));
    expect(users).toHaveLength(1);
    expect(users[0].email).toBe('john.doe@example.com');
  });
});




                  // Test for OrderCard component
    test('renders OrderCard component without crashing', () => {
        render(
            <Router>
                <ThemeProvider theme={theme}>
                    <OrderCard />
                </ThemeProvider>
            </Router>
        );
    });
});
  

  // Test for PaymentOptions component
  test('renders PaymentOptions component without crashing', () => {
    render(
        <Router>
            <PaymentOptions />
        </Router>
    );
});
