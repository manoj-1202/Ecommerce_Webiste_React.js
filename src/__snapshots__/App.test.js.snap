// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Tests matches snapshot for Banner component 1`] = `
<div
  className="MuiBox-root css-1w014vv"
>
  <img
    alt="Banner"
    className="css-c64u4s"
    src="images/banner/banner.png"
  />
  <div
    className="MuiBox-root css-1apetot"
  >
    <h5
      className="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
    >
      Huge Collection
    </h5>
    <h2
      className="MuiTypography-root MuiTypography-h2 css-ik58jw-MuiTypography-root"
    >
      New Bags
    </h2>
    <h6
      className="MuiTypography-root MuiTypography-subtitle1 css-1f8kc1i-MuiTypography-root"
    >
      Discover our collection of stylish bags, from sleek leather designs to versatile travel essentials, perfect for any occasion and designed to meet your every need.
    </h6>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1kkjskx-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Shop Now
    </button>
  </div>
</div>
`;
